src/tests/jsonLogic.test.js

```diff
@@ -322,6 +326,74 @@ describe('cross-value validations', () => {
     });
   });
 
+  describe('Conditionals', () => {
+    it('when field_a > field_b, show field_c', () => {
+      const { fields, handleValidation } = createHeadlessForm(
+        schemaWithGreaterThanChecksForThreeFields,
+        { strictInputType: false }
+      );
+      expect(fields.find((i) => i.name === 'field_c').isVisible).toEqual(false);
+
+      expect(handleValidation({ field_a: 1, field_b: 3 }).formErrors).toEqual(undefined);
+      expect(handleValidation({ field_a: 1, field_b: null }).formErrors).toEqual({
+        field_b: 'Required field',
+      });
+      expect(handleValidation({ field_a: 10, field_b: 3 }).formErrors).toEqual({
+        field_c: 'Required field',
+      });
+      expect(handleValidation({ field_a: 10, field_b: 3, field_c: 0 }).formErrors).toEqual(
+        undefined
+      );
+    });
+
+    it('A schema with both a `x-jsf-validations` and `properties` check', () => {
+      const { handleValidation } = createHeadlessForm(
+        schemaWithPropertiesCheckAndValidationsInAIf,
+        { strictInputType: false }
+      );
+      expect(handleValidation({ field_a: 1, field_b: 3 }).formErrors).toEqual(undefined);
+      expect(handleValidation({ field_a: 10, field_b: 3 }).formErrors).toEqual({
+        field_c: 'Required field',
+      });
+      expect(handleValidation({ field_a: 5, field_b: 3 }).formErrors).toEqual(undefined);
+    });
+
+    it('Conditionally apply a validation on a property depending on values', () => {
+      const { fields, handleValidation } = createHeadlessForm(
+        schemaWithChecksAndThenValidationsOnThen,
+        { strictInputType: false }
+      );
+      const cField = fields.find((i) => i.name === 'field_c');
+      expect(cField.isVisible).toEqual(false);
+      expect(cField.description).toEqual(undefined);
+      expect(handleValidation({ field_a: 10, field_b: 5 }).formErrors).toEqual({
+        field_c: 'Required field',
+      });
+      expect(handleValidation({ field_a: 10, field_b: 5, field_c: 0 }).formErrors).toEqual({
+        field_c: 'Needs more numbers',
+      });
+      expect(cField.description).toBe('I am a description!');
```

Comment:
what a joy!!

Reply:
ah but it's missing the else. it's a edge-case that happened in the past where you trigger `then`, but if you trigger else again, it get's stuck on `then`. Could you add a last one with a: 5; b: 10; and assert the description is null and the field is invisible?

Reply:
I can do the isVisible part easy, but I'm finding the description part tricky. It always comes back with `I am a description!` because it doesn't know _where_ to update. I don't think also fields should be wiped if they aren't visible 🤔 

Gonna pick away it a bit more.

Reply:
I believe that bug is related to [this internal issue](https://linear.app/remote/issue/RMT-58/jsf-conditionals-keywords-are-redundant-eg-readonly-false)... if it is, then leave it as it is (maybe with the assertions commented out and a code comment mentioning the issue id (RMT-58)

Reply:
Yeah, you're dead right - exact same thing. added small note in ee18568ba2edc9717fda8ae1c3e59058386f5f07